lib/Sema/CSDiagnostics.h

```diff
@@ -98,7 +98,7 @@ class FailureDiagnostic {
   /// Resolve type variables present in the raw type, if any.
   Type resolveType(Type rawType, bool reconstituteSugar = false,
                    bool wantRValue = true) const {
-    auto resolvedType = CS.simplifyType(rawType);
+    auto resolvedType = CS.simplifyType(rawType, /*forDiagnostics*/true);
```

Comment:
Since all the logic involved in diagnostics goes through `resolveType` I think it might makes more sense just "re-implement" `simplifyType` here the way we want to keep constraint system in the dark about how we want "holes" printed. WDYT?

Reply:
Sure, I'm happy to do that. It did feel weird to put something specific to diagnostics into `simplifyType`, but I needed the recursive behavior. I'll make that change.